projects/vdk-core/src/vdk/internal/builtin_plugins/job_secrets/datajobs_service_secrets.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+from copy import deepcopy
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Union
+
+from vdk.api.job_input import ISecrets
+from vdk.api.plugin.plugin_input import ISecretsServiceClient
+
+from ...core.errors import log_and_throw
+from ...core.errors import ResolvableBy
+from .base_secrets_impl import check_valid_secret
+
+log = logging.getLogger(__name__)
+
+SecretValue = Union[int, float, str, list, dict, None]
+
+
+class DataJobsServiceSecrets(ISecrets):
+    """
+    Data Jobs Secrets implementation.
+    """
+
+    __VALID_TYPES = [int, float, str, list, dict, type(None)]
+
+    def __init__(
+        self,
+        job_name: str,
+        team_name: str,
+        secrets_service_client: ISecretsServiceClient,
+        write_preprocessors: Optional[List[ISecretsServiceClient]] = None,
```

Comment:
Could you give an example of what a preprocessor might do.
Just curious.

Reply:
Its too confusing using one interface for different purposes.

Reply:
I believe this is part of the extensibility mechanism which allows users to plug-in write pre-processor which modify secrets or properties before they are stored.